osu.Server.Spectator.Tests/MultiplayerFlowTests.cs

```diff
@@ -766,6 +767,52 @@ public async Task RoomSettingsUpdateNotifiesOtherUsers()
             mockReceiver.Verify(r => r.SettingsChanged(testSettings), Times.Once);
         }
 
+        [Fact]
+        public async Task ChangingSettingsToNonExistentBeatmapThrows()
+        {
+            mockDatabase.Setup(d => d.GetBeatmapChecksumAsync(3333)).ReturnsAsync((string)null!);
```

Comment:
While the database does seem to list the `checksum` column as nullable, that would mean that the return type of `GetBeatmapChecksumAsync()` should *actually* be `Task<string?>`. Because note that `DatabaseAccess` uses `Single()`, not `SingleOrDefault()`. So a null here would have to physically reside *in the database*.

I'm not sure if there is a case of a null checksum in production, but either way - either this test should be removed due to testing an unrealistic scenario, or the types should be adjusted to avoid this `null!` nonsense.

This is all setting aside the fact of adding tests unrelated to the actual goal of this pull.

Reply:
> While the database does seem to list the checksum column as nullable, that would mean that the return type of GetBeatmapChecksumAsync() should actually be Task<string?>. Note that DatabaseAccess uses Single(), not SingleOrDefault(). So a null here would have to physically reside in the database.

Yeah, I was a bit hesitant about it, but got too busy to look through it, my bad though. I've went ahead and made `GetBeatmapChecksumAsync()` return nullable instead. 

Not sure about straight up changing it to `SingleOrDefault()` though, @peppy can you confirm whether you intended making `GetBeatmapChecksumAsync()` use `QuerySingleAsync()` instead of `QuerySingleOrDefaultAsync()`, sounds like it should've been the latter?

> This is all setting aside the fact of adding tests unrelated to the actual goal of this pull.

I have added these tests while I was at this, don't see any complexity to move it into its own PR. (not to mention that I wasn't really comfortable at all with adding just one only test for the spotted edge case)